mava/wrappers/env_preprocess_wrappers.py

```diff
@@ -273,3 +274,145 @@ def __getattr__(self, name: str) -> Any:
             return self.__getattribute__(name)
         else:
             return getattr(self._environment, name)
+
+
+class StackObservations:
+    """Stack frames of observations together"""
+
+    def __init__(self, environment: Any, num_frames: int = 1):
+        """Initialise wrapper."""
+
+        self._environment = environment
+
+        # Check that this is the first wrapper that is called on the environment
+        if self._environment.obs_normalisation_start_index != 0:
+            raise ValueError(
+                "Observation stacking should be the first cumstom\
+                    wrapper we call in the environment factory."
+            )
+
+        self.frames: Any = {
+            agent: deque([], maxlen=num_frames)
+            for agent in self._environment.possible_agents
+        }
+        self.num_frames = num_frames
+
+    def reset(self) -> dm_env.TimeStep:
+        """Reset the environment by setting the same observation to all the frames"""
+
+        timestep = self._environment.reset()
+        if type(timestep) == tuple:
+            timestep, env_extras = timestep
+        else:
+            env_extras = {}
+
+        new_observations: Any = {}
+
+        agent_0 = self._environment.possible_agents[0]
+        if type(timestep.observation[agent_0]) == OLT:
```

Comment:
Can you add docs for this if statement

Reply:
Thanks @OmaymaMahjoub. I doubt this is still necessary actually. The first type I wrote this code, we had environments whose observations were not OLT named tuples so I used this if statement to cover both cases. I will remove this and see the tests passes.

Reply:
I was wandering why we have this if statement in this class and not in the ConcatAgentIdToObservation and ConcatPrevActionToObservation too?

Reply:
I have removed it and everything works now. Initially this use to throw issues in the wrappers test. But everything was trimmed recently and now we don't need this. Also the test I commented in the TODO now works. Also it used to be a problem because of the different observation types

Reply:
Okay!